src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -205,6 +206,24 @@ private async Task InitializeAsync()
             var solutionLoadedGuid = VSConstants.UICONTEXT.SolutionExistsAndFullyLoaded_guid;
             _vsMonitorSelection.GetCmdUIContextCookie(ref solutionLoadedGuid, out _solutionLoadedUICookie);
 
+            if (ErrorHandler.Succeeded(_vsSolution.GetProperty((int)__VSPROPID.VSPROPID_IsSolutionOpen, out object isSoltuionOpen)))
+            {
+                _isSolutionOpen = (bool)isSoltuionOpen;
+            }
+            else
+            {
+                _isSolutionOpen = false;
+            }
+
+            if (ErrorHandler.Succeeded(_vsSolution.GetProperty((int)__VSPROPID.VSPROPID_SolutionDirectory, out object solutionDirectory)))
+            {
+                _solutionDirectory = (string)solutionDirectory;
+            }
+            else
+            {
+                _solutionDirectory = null;
```

Comment:
No. I should probably have copied the same comment from https://github.com/NuGet/NuGet.Client/pull/4879/files/fd029a7386b50328354b64c2e9d603712aa542ed#diff-bed53a8b2e737e49ff6141c4279e525dfa2d100eabf6510ef6e0a1746de10505R1139

This was the exact reason that the "temporary solution" test was failing on previous iteration of the PR.

In theory the inverse could be true. When `IsSolutionOpen` is false, then `SolutionDirectory` is probably always true. But I don't know that this is actually correct. There's no docs stating this. The `SolutionDirectory == null` while `IsSolutionOpen == true` is non-intuitive, and also not documented. So, I was playing it safe by initializing the two values independently.

Although, writing all this makes me realize that if `IsSolutionOpen == false` and `SolutionDirectory != null` is a possible scenario, then this PR will introduce a bug, because we will never determine what the solution directory is when the solution is closed (but my gut feeling is that it is an impossible scenario).

However, I do need to fix the "solution save as" handler to fix the solution directory value.

Reply:
> However, I do need to fix the "solution save as" handler to fix the solution directory value.

Thank you for pushing a new commit for this case.